app/src/main/java/io/homeassistant/companion/android/controls/ClimateControl.kt

```diff
@@ -49,12 +49,21 @@ class ClimateControl {
                     else -> entity.state
                 }
             )
+            val minValue = (entity.attributes["min_temp"] as? Number)?.toFloat() ?: 0f
+            val maxValue = (entity.attributes["max_temp"] as? Number)?.toFloat() ?: 100f
```

Comment:
`min_temp` and `max_temp` on my nest thermostat are used differently.  My target temperature is 70 so min and max are both 70.  This would break the integration for.

Reply:
oh, I was afraid of that....hmm so not sure how we can proceed with fixing #1217 its more a usability thing than an actual bug.

Reply:
Perhaps we can get what units are set in HA and then use a better range?

Reply:
Yea I will do 2 things in the next update.

1. We will use `unit_system` as defined in https://github.com/home-assistant/core/blob/dev/homeassistant/components/mobile_app/webhook.py#L553
2. I will reference the thermostat card and see if we can use the same ranges defined there to make it match the front end more
3. Probably use some sane defaults if step 2 doesn't work like I think it will, so maybe best to stick to 5-40 for Celsius and 40-90 for Fahrenheit

Reply:
Looks like the `climate` integration itself uses a Celsius range....not sure if these defaults make sense either when compared to the users expectation.

https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/__init__.py#L75

the thermostat card itself goes by `target_temp_low` which also not all thermostats support (like mine) LOL so we may need to stick to sane defaults in this case.

https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/cards/hui-thermostat-card.ts#L125

Reply:
Maybe I missed a point, but wouldn't be a good solution to use the PR's current logic, and fall back to some default range (with the new logic of using the `unit_system` for defaults), when the current temperature is equal to the min and max temperature, like in case of JBassett?

Reply:
Am I misunderstanding something around the official HA dev docs around climate or is it suggesting that `nest` is not using these attributes appropriately?

https://developers.home-assistant.io/docs/core/entity/climate#properties

The docs do not mention that these attributes must be tied to the current set point of the thermostat.  It sounds like `min_temp` and `max_temp` are more device specific ranges and are not meant to be a set point specific range like the target temps suggest.

Reply:
After looking over the documentation I think you are correct.  They should be using `target_temperature_high` and `target_temperature_low`.  If we get a defect in we might be able to get it fixed before the next release of core then we can merge you change.

Reply:
Scratch that, it's because I have my thermostat locked (kids find it fun to spin the dial, but my bill doesn't find it funny).  Because I have it locked at a single temperature that's what's returned.